openshift/README.md

```diff
@@ -1,33 +1,49 @@
-# Deploying fabric8-analytics services
+# Openshift green field deployment of fabric8-analytics services
 
-## Create a new project
+Config map and secrets are generated from the configuration stored in environment variables.
 
-`oc login` to your favorite OpenShift cluster and create a new project there:
+## Configure fabric8-analytics services
+All configuration for the deployment script resides in env.sh.
+Create file env.sh inside this directory with following contents.
+This script is ignored by git so feel free to change it locally.
 
-```
-oc new-project fabric8-analytics
-```
+`export OC_URI='dev.rdu2c.fabric8.io:8443'`
 
-## Deploy fabric8-analytics services
+`export OC_USERNAME='Not set'`
+
+`export OC_PASSWD='Not set'`
+
+`export OC_PROJECT=$(whoami)'-greenfield-test'`
+
+`export AWS_ACCESS_KEY_ID='Not set'`
+
+`export AWS_SECRET_ACCESS_KEY='Not set'`
+
+`export GITHUB_API_TOKENS='Not set'`
+
+`export GITHUB_OAUTH_CONSUMER_KEY='Not set'`
 
-Generate and deploy [config map](./README-ConfigMap.md) first:
+`export GITHUB_OAUTH_CONSUMER_SECRET='Not set'`
 
-```
-./generate-config.sh  # will create config.yaml file
-oc apply -f config.yaml
-```
+`export FLASK_APP_SECRET_KEY='Not set'`
 
-Deploy secrets:
+`export RDS_ENDPOINT=''`
 
-```
-oc process -f secrets-template.yaml -v AWS_ACCESS_KEY_ID='..' -v AWS_SECRET_ACCESS_KEY='...' -v GITHUB_API_TOKENS='...' -v GITHUB_OAUTH_CONSUMER_KEY='...' -v GITHUB_OAUTH_CONSUMER_SECRET='...' -v FLASK_APP_SECRET_KEY='...' RDS_ENDPOINT='...' RDS_PASSWORD='...' | oc apply -f -
-```
+`export RDS_INSTANCE_NAME="$OC_USERNAME-bayesiandb"`
 
-Note all secrets need to be base64 encoded:
+`export RDS_INSTANCE_CLASS='db.t2.micro'`
 
-```
-echo -n 'my-secret-key' | base64 -w 0
-```
+`export RDS_DBNAME='postgres'`
+
+`export RDS_DBADMIN='coreapi'`
+
+`export RDS_STORAGE=5`
+
+`export RDS_PASSWORD='somethingclever'`
```

Comment:
We should make sure that people will actually change this password. Maybe check it in `deploy.sh` and fail if they try to use this default value?